### DISCUSSION TITLE
Is it possible to retrieve only FINAL candle data, not continuous updates?

### DISCUSSION THREAD

**Discussion by am11y** [oliver-zehentleitner/unicorn-binance-websocket-api#280] [General]
Hi,
is it possible to subscribe to binance kline streams and receive only FINAL candle updates, at the time when candle is closed?

Not that I receive a kline update each time a trade happens; for example if I subscribe to 1 minute stream, I want to get kline update only when full minute is over.

**Comment by am11y** [comment]
To clarify: I meant if Binance **itself** ofers such a stream, that new data is pushed only at the closing time of a candle?

Not that I filter it myself with script by checking the 
"x": false,     // Is this kline closed?

property.

**Comment by oliver-zehentleitner** [comment]
to my knowledge: No :)

**Comment by molivramento** [comment]
Faça algo como:

...
o = None
while True:
res = await klines.recv()
if res['k']['o'] != o:
o = res['k']['o']
print(res)

isso vai garantir que os dados sejam impressos apenas quando o valor "aberto" for alterado. (apenas a logica, adapte-o para unicornio)

**Reply by oliver-zehentleitner** [reply]
I cant read what you wrote but here is some info about how to get the latest kline: https://github.com/LUCIT-Systems-and-Development/unicorn-binance-websocket-api/discussions/270#discussioncomment-2720291
